File: c#/prototype/Ejm1/AccountPrototypeFactory.cs

namespace Prototype.Ejm1
{
    /// <summary>
    /// Esta clase es definida para contener objetos prototipo que representan cuentas UserAccount 
    /// de tipo Supervisor y AccountRep
    /// </summary>
    public class AccountPrototypeFactory
    {
        private UserAccount AccountRep;
        private UserAccount Supervisor;

        public AccountPrototypeFactory(UserAccount supervisorAccount,
                UserAccount arep)
        {
            AccountRep = arep;
            Supervisor = supervisorAccount;
        }

        public UserAccount GetAccountRep()
        {
            return (UserAccount)AccountRep.Clone();
        }

        public UserAccount GetSupervisor()
        {
            return (UserAccount)Supervisor.Clone();
        }
    }
}

File: java/prototype/ej1/AccountPrototypeFactory.java
package patrones.prototype.ej1;

/**
 * Esta clase es definida para contener objetos prototipo que representan cuentas UserAccount de tipo Supervisor y AccountRep
 *
 */
public class AccountPrototypeFactory {
	private UserAccount accountRep;
	private UserAccount supervisor;

	public AccountPrototypeFactory(UserAccount supervisorAccount,
			UserAccount arep) {
		accountRep = arep;
		supervisor = supervisorAccount;
	}

	public UserAccount getAccountRep() {
		return (UserAccount) accountRep.clone();
	}

	public UserAccount getSupervisor() {
		return (UserAccount) supervisor.clone();
	}
}
